x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -186,12 +187,14 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
     objects: Array<SavedObjectsBulkUpdateObject<T>> = [],
     options: SavedObjectsBaseOptions = {}
   ) {
-    await this.ensureAuthorized(
-      this.getUniqueObjectTypes(objects),
-      'bulk_update',
-      options && options.namespace,
-      { objects, options }
-    );
+    const objectNamespaces = objects
+      .filter(({ namespace }) => namespace !== undefined)
+      .map(({ namespace }) => namespace!);
+    const namespaces = uniq([namespaceIdToString(options?.namespace), ...objectNamespaces]);
```

Comment:
Ah, I wrote it this way because I didn't want to change the method signature for `ensureAuthorized` and `checkPrivileges` (arg: `namespaceOrNamespaces?: string | string[]`). But it sounds like we are leaning away from exposing `namespaceIdToString` and `namespaceStringToId` from core, so I'll make this change.

Reply:
> Ah, I wrote it this way because I didn't want to change the method signature for ensureAuthorized and checkPrivileges (arg: namespaceOrNamespaces?: string | string[]). But it sounds like we are leaning away from exposing namespaceIdToString and namespaceStringToId from core, so I'll make this change.

Ah, I failed to realize this was `string | string[]` and not `string | Array<string | undefined>` when I made this proposal.

If the work to change the signatures turns out to be a ton of work, then let's revisit. It probably won't help with anything other than readability, but we could consider adding a couple of types to help. Something like

```ts
type SavedObjectsNamespace = string;
type SavedObjectsRawNamespace = string | undefined;
```

And then we could have functions to convert to/from raw.

I don't love the term "raw" to describe this, so open to other ideas (or simply ignore this idea)

Reply:
Done in 440d9877.

The blast radius was pretty small, I don't think we need new types.